modules/etcd/ec2.tf

```diff
@@ -1,3 +1,30 @@
+resource "aws_ebs_volume" "etcd" {
+  count = "${ length( split(",", var.azs) ) }"
+
+  availability_zone = "${ element( split(",", var.azs), 0 ) }"
+
+  type = "gp2"
+  size = 100
+
+  tags {
+    builtWith = "terraform"
+    Cluster = "${ var.name }"
+    depends-id = "${ var.depends-id }"
+    KubernetesCluster = "${ var.name }"
+    Name = "etcd${ count.index + 1 }-${ var.name }"
+    role = "etcd,apiserver"
+    version = "${ var.coreos-hyperkube-tag }"
+  }
+}
+
+resource "aws_volume_attachment" "etcd" {
+  count = "${ length( split(",", var.azs) ) }"
+
+  device_name = "/dev/xvdf"
```

Comment:
I forgot to include some changes to cloud-config.tf. On startup we format /dev/xvdf and mount it to /media/etcd2 and instruct etcd to use that location to persist state. I chose xvdf because that's the device we're using on worker nodes for ephemeral storage. See 8e73f42.